subsys/bluetooth/host/smp.c

```diff
@@ -264,6 +265,44 @@ static bool sc_supported;
 static const u8_t *sc_public_key;
 static K_SEM_DEFINE(sc_local_pkey_ready, 0, 1);
 
+int bt_smp_get_pairing_peer_info(const struct bt_conn *conn,
+				 struct bt_smp_pairing *info)
+{
+	int i;
+
+	BT_DBG("conn %p handle %u", conn, conn->handle);
+
+	if (conn == NULL) {
+		return -EINVAL;
+	}
+
+	for (i = 0; i < ARRAY_SIZE(bt_smp_pool); i++) {
+		struct bt_smp *smp = &bt_smp_pool[i];
+
+		if (smp->chan.chan.conn != conn) {
+			continue;
+		}
+
+		if (!atomic_test_bit(smp->flags, SMP_FLAG_USER)) {
+			return -EBUSY;
+		}
+
+		if (IS_ENABLED(CONFIG_BT_PERIPHERAL) &&
+		    atomic_test_bit(smp->flags, SMP_FLAG_PAIRING)) {
+			*info = *(struct bt_smp_pairing *)&smp->preq[1];
+		} else if (IS_ENABLED(CONFIG_BT_CENTRAL) &&
+			   atomic_test_bit(smp->flags, SMP_FLAG_SEC_REQ)) {
+			*info = *(struct bt_smp_pairing *)&smp->prsp[1];
```

Comment:
This is the pairing response, which is not guaranteed to have been set at the same time as the lifetime of the `SMP_FLAG_SEC_REQ`. This flag is set when having received the Security Request, we need the Pairing Response.
Also security request is optional so this might not be set at all.

The flag `SMP_FLAG_PAIRING` is set by the central when it is sending the pairing_request, so that is not correct either.
I think we either have to do `if(atomic_test_bit(smp->flags, SMP_FLAG_PAIRING) && !atomic_test_bit(&smp->allowed_cmds, BT_SMP_CMD_PAIRING_RSP))`, which seems somewhat convoluted. So maybe the best option is to have a new flag which is set when having received the pairing response in `smp_pairing_rsp`.